src/main/thrift/parquet.thrift

```diff
@@ -812,8 +837,20 @@ struct ColumnMetaData {
 
   /** Set of all encodings used for pages in this column chunk.
    * This information can be used to determine if all data pages are
-   * dictionary encoded for example **/
+   * dictionary encoded for example  
+   *
+   *  PAR1: Optional. May be deprecated in a future release in favor
+   *        serialized_encoding_stats.
+   *  PAR3: Don't populate.  Write serialized_page_encoding_stats.
+   **/
   13: optional list<PageEncodingStats> encoding_stats;
+  /** 
+    * Serialized page encoding stats.
+    *
+    * PAR1: Start populating after encoding_stats is deprecated.
+    * PAR3: Populate instead of encoding_stats.
+    */
+  17: optional binary serialized_encoding_stats
```

Comment:
I think it makes sense to adopt @alkis's idea of using a boolean field here to indicate whether dictionary encoding is applied to all data pages.

Reply:
Sorry I just saw line 943. I meant field 17 is not required any more.

Reply:
But I just got this implemented 😭 😉. Joking aside, is there any other use for this field beyond optimizing dictionary based queries?

Reply:
Yes I think it serves three purposes:
1.  consolidate encodings.
2. Determine optimizations available in the engine (e.g. fully dictionary encoded, fully RLE, etc)
3. Instrumentation for engines to understand what the distribution of data they see to prioritize optimizations for specific encodings.

I'm OK removing completely but my concern around #2 is spec changes take a while to propagate, having the information for engines to determine what optimizations they want to do without requiring a new flag I think is useful.

Reply:
If `ColumnMetaData::encodings` could distinguish between PLAIN encoded dictionary and PLAIN data encodings, the detection of full dictionary encoding could be based on that without page specific statistics. If `Encoding::PLAIN_DICTIONARY` could be un-deprecated and indicate a PLAIN encoded dictionary page then `encoding_stats` might not be needed for most usecases. A list of `[PLAIN_DICTIONARY, RLE_DICTIONARY]` could indicate full dictionary encoding for the column.

Reply:
yes, I thought I had marked encodings as dropped (as the information here duplicates it at the very least), if not I'll update it.  I'm also OK with readers discovering unreadable pages lazily.

Reply:
@jhorstmann this seems plausible however, I still think there is use in this data from a debuggability standpoint to elaborate on #3 I shouldn't have just limited it to engines.  Being able to dump encodings and how many there are quickly allows users to understand a little bit better how there data is being compressed and let them potentially fine tune these aspects.  It isn't a frequent occurrence but there is certainly cases when having a little bit more data helps with debugability.